posts/inside-rust/2022-08-11-diagnostic-effort.md

```diff
@@ -0,0 +1,336 @@
+to produce output in languages other than English.
+
+For example, consider the following diagnostic where a user has used a colon to
+specify a function's return type instead of an arrow:
+
+```text
+error: return types are denoted using `->`
+ --> src/main.rs:1:21
+  |
+1 | fn meaning_of_life(): u32 { 42 }
+  |                     ^ help: use `->` instead
+```
+
+We could output that diagnostic in Chinese:
+
+```text
+错误: 返回类型使用`->`表示
+ --> src/main.rs:1:21
+  |
+1 | fn meaning_of_life(): u32 { 42 }
+  |                     ^ 帮助: 使用`->`来代替
+```
+
+or even in Spanish:
+
+```text
+error: el tipo de retorno se debe indicar mediante `->`
+ --> src/main.rs:1:21
+  |
+1 | fn meaning_of_life(): u32 { 42 }
+  |                     ^ ayuda: utilice `->` en su lugar
+```
+
+Translated error messages will allow non-native speakers of English to use Rust
+in their preferred language.
+
+## What's the current status?
+Implementation on diagnostic translation has started, but we're looking for
+help!
+
+The core infrastructure for diagnostic translation has been implemented in
+`rustc`; this makes it possible for Rust to emit a diagnostic with translated
+messages. However, every diagnostic in `rustc` has to be ported to use this new
+infrastructure, otherwise they can't be translated. That's a lot of work, so
+the diagnostics working group has chosen to combine the translation effort with
+a transition to "diagnostic structs" (more on that later) and get both done at
+once.
+
+Once most diagnostic messages have been ported to the new infrastructure, then
+the diagnostics working group will start creating a workflow for translation
+teams to translate all of the diagnostic messages to different languages.
+
+Every pull request related to diagnostic translation is listed [in this
+document](https://hackmd.io/@davidtwco/rkXSbLg95).
+
+## Getting involved
+There's a lot of work to do on diagnostic translation but the good news is that
+lots of the work can be done in parallel and it doesn't require background in
+compiler development or familiarity with `rustc` to contribute!
+
+If you are interested, feel free to just get started! You can ask for help in
+[`#t-compiler/wg-diagnostics`] or reach out to [`@davidtwco`].
+
+**Note:** This post isn't going to be updated as the working group iterates on
+and improves the workflow for diagnostic translation, so always consult the
+developer guide for the most recent documentation on [diagnostic
+structs][diag_struct] or [diagnostic translation][diag_translation].
+
+### 1. Setting up a local development environment
+Before helping with the diagnostic translation effort, you'll need to get your
+development environment set up, so [follow the instructions on the `rustc` dev
+guide][getting_started].
+
+### 2. Getting ready to port your first diagnostic
+Almost all diagnostics in `rustc` are implemented using the traditional
+`DiagnosticBuilder` APIs, which look something like this:
+
+```rust=
```

Comment:
what does `rust=` mean

Reply:
In some markdown renderers this adds line numbers in addition to syntax highlighting. Not sure if it does here but seems unnecessary even if it does.

Reply:
Fixed, think this was automatic from HackMD.